recipe/build-mpi.sh

```diff
@@ -1,27 +1,21 @@
 #!/bin/bash
 
-# configure balks if F90 is defined
-# with a fatal deprecation message pointing to FC
+# unset unused old fortran compiler vars
 unset F90 F77
 
-# remove --as-needed, which causes problems for downstream builds,
-# seen in failures in petsc, slepc, and hdf5 at least
-export LDFLAGS="${LDFLAGS/-Wl,--as-needed/}"
```

Comment:
Can you explain why this fix was removed?

Reply:
Because it is no longer needed. After you made this fix, later on, you updated the build script to use the various `MPICHLIB_xFLAGS`, and redefine minimum `xFLAGS` with basic prefix awareness. I don't thing this is the cause of the failures in Linux. I'll elaborate in another comment.

Anyway, MPICH's configure completely ignores `MPICHLIB_LDFLAGS`. Just grep for it in configure, not used at all! I guess this should be reported upstream. Being able to pass `-Wl,--as-needed` in  `MPICHLIB_LDFLAGS` is definitely a good thing, so we can remove bogus library dependencies.

Reply:
Gotcha, thanks!